shelley/chain-and-ledger/shelley-spec-ledger-test/test/Test/Shelley/Spec/Ledger/Examples/Updates.hs

```diff
@@ -194,9 +194,51 @@ expectedStEx1 =
     . C.newLab blockEx1
     . C.feesAndDeposits feeTx1 (Coin 0)
     . C.newUTxO txbodyEx1
-    . C.setCurrentProposals ppVotes1
+    . setCurrentProposals ppVotes1
```

Comment:
I'm curious, is there a reason for moving this function out of the module it was in?

Reply:
Good question :) The module this function was in had era independent function, however `setCurrentProposals` is now specific to the PPUP state, hence I thought it'd fit better in the `Examples.Update` module, since it is Shelley specific. Does that make sense?

Reply:
ah I see. I feel like it really belongs with the other things that mutate the chain state. the `Combinators` module is like a utility module, and the `Updates` module is more like a specific example. maybe we could give it a name that makes it clear that it is for shelley only?

Reply:
> I feel like it really belongs with the other things that mutate the chain state.

So the `Combinators` module seemed to me era independent. When I abstracted the update state, this function became era dependent. So that's why I moved to the only module that used it, which was Shelley specific.

> maybe we could give it a name that makes it clear that it is for shelley only?

Sure. Do you have a suggestion? `setCurrentShelleyUpdateProposals`?

Reply:
The other issue is that it's not so much shelley specific at this point, since we want this to also work for Allegra, Mary and Alonzo. We can do this with appropriate constraints on the relevant parts of the era.